hack/common.sh

```diff
@@ -128,9 +132,18 @@ os::build::build_binaries() {
     for platform in "${platforms[@]}"; do
       os::build::set_platform_envs "${platform}"
       echo "++ Building go targets for ${platform}:" "${targets[@]}"
-      go install "${goflags[@]:+${goflags[@]}}" \
-          -ldflags "${version_ldflags}" \
-          "${binaries[@]}"
+      for b in ${binaries[@]}; do
+          echo "Building $b"
+          if [[ "$b" == *"e2e.test"* ]]; then
```

Comment:
although i like explicit calculation of the build package and the test output location for simplicity, i think this could be generalized.  open to opinions here.  this build script is pretty complicated so i figured id just do something as explicit as possible without hardcoding any paths.